### PR TITLE
Feature/component

### PR DESCRIPTION
**Pull request by juliusb0205**
@ming-ancino 

Worked on 1, 2, 3

### CODE REVIEW

**Comment by juliusb0205** [comment]
@ming-ancino 

removed duplicate Newton.Component
moved tests to test.js

**Comment by juliusb0205** [comment]
@ming-ancino 

Moved guid closure to component

**Comment by juliusb0205** [comment]
@ming-ancino 

changed for in to for-loops and Object.each
transfer misplaced test

**Comment by juliusb0205** [comment]
@ming-ancino 

Updated

File: js/component.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    return null;
+  }
+
+  self.prototype.getInitialState = function(){
+    return {};
+  }
+
+  self.prototype.setState = function(newState){
+
+    for ( var key in newState ) {
```
Comment (ming-ancino) [inline js/component.js]: wsas

File: js/component.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      if ( newState.hasOwnProperty(key) ) {
```
Comment (ming-ancino) [inline js/component.js]: sdfdsad

File: js/component.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        this.state[key] = newState[key];
+      }
+    }
+
+  }
+
+  namespace.createClass = function(props){
+
+    for ( var key in props ) {
```
Comment (ming-ancino) [inline js/component.js]: asdasd

File: js/component.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      if ( props.hasOwnProperty(key) ) {
```
Comment (ming-ancino) [inline js/component.js]: asdasd

File: js/element.js [hunk 5]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
 
 (function(namespace){
 
-  var self = namespace.Element = function(type, attributes){
+  var self = namespace.Element = function(type, attributes) {
```
Comment (ming-ancino) [inline js/element.js]: asdadada

File: js/element.js [hunk 6]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
-        }
-      }
-    }
 
-    return elem;
-  }
 
-  self.prototype.setChildren = function(children){
+
+  self.prototype.setChildren = function(children) {
```
Comment (ming-ancino) [inline js/element.js]: qweqwe

File: js/element.js [hunk 7]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
-                 {className: 'outer-div'});
 
-  var inner_div = new Newton.Element('div',
-                 {className: 'inner-div'});
 
-  var p = new Newton.Element('p', null);
-  p.setChildren(['Hello World']);
-  inner_div.setChildren([p]);
-  outer_div.setChildren([inner_div]);
+  self.prototype._setAttributes = function(elem) {
```
Comment (ming-ancino) [inline js/element.js]: asdadad

File: js/element.js [hunk 8]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
+    var obj = this.attributes;
+    for ( var key in obj ) {
```
Comment (ming-ancino) [inline js/element.js]: asdadad

File: js/element.js [hunk 9]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
+      if ( obj.hasOwnProperty(key) ){
+        elem[key] = obj[key];
+      }
+    }
+  }
+
+  self.prototype._renderChildren = function(elem){
+    var child = this.children;
+    for ( var i = 0; i < child.length; i++ ) {
```
Comment (ming-ancino) [inline js/element.js]: asdasdad

File: js/element.js [hunk 10]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
+      if ( child[i] instanceof Newton.Element ) {
```
Comment (ming-ancino) [inline js/element.js]: asdasd

File: js/element.js [hunk 11]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
+        elem.appendChild ( child[i].render() );
```
Comment (ming-ancino) [inline js/element.js]: asdadadad

File: js/element.js [hunk 12]
```diff
@@ -2,50 +2,50 @@ var Newton = Newton || {};
...
+      }
+      else {
+        elem.appendChild ( document.createTextNode(child[i]) );
```
Comment (ming-ancino) [inline js/element.js]: asddad

**Pull request merged**